### PR TITLE
Changes refactor

### PR DESCRIPTION
**Pull request by czue**
goes with https://github.com/dimagi/pillowtop/pull/147

also removes some dead pillows and adds some tests

best read commit-by-commit

### CODE REVIEW

File: testapps/test_pillowtop/tests/test_settings.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import os
+from django.conf import settings
+from django.test import TestCase, SimpleTestCase
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_settings.py]: F401 'SimpleTestCase' imported but unused

**Comment by emord** [comment]
@czue was rebuilding staging but this branches pillowtop pointed to a commit that doesn't exist. Guessing you were on a local branch when you pushed? I changed it to the head of https://github.com/dimagi/pillowtop/pull/147

**Comment by czue** [comment]
ah, nice thanks and sorry - must have rebased and forgotten to update the submodule

File: testapps/test_pillowtop/tests/test_settings.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import os
+from unittest import skip
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_settings.py]: F401 'skip' imported but unused

File: testapps/test_pillowtop/tests/test_settings.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+from django.conf import settings
+from django.test import TestCase
+import json
+from corehq.util.test_utils import TestFileMixin
+from pillowtop import get_all_pillow_instances, get_all_pillow_classes
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_settings.py]: F401 'get_all_pillow_instances' imported but unused

File: testapps/test_pillowtop/tests/test_require_elastic.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        @require_elasticsearch
+        def rawes_fail():
+            rawes.Elastic('example.com:9999').get('nothing_here', timeout=0.1)
+        rawes_fail()
+
+    def test_no_fail_on_elasticsearch_errors(self):
+        @require_elasticsearch
+        def elasticsearch_fail():
+            es = Elasticsearch([{
+                    'host': 'example.com',
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_require_elastic.py]: E126 continuation line over-indented for hanging indent

File: testapps/test_pillowtop/tests/test_require_elastic.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                    'port': 9999,
+                }],
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_require_elastic.py]: E121 continuation line under-indented for hanging indent

File: testapps/test_pillowtop/tests/test_require_elastic.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                timeout=0.1,
+            )
+            es.info()
+        elasticsearch_fail()
+
+
+
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/test_require_elastic.py]: W391 blank line at end of file

File: testapps/test_pillowtop/tests/utils.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    @functools.wraps(fn)
+    def decorated(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except (requests_exceptions.ConnectionError, elasticsearch_exceptions.ConnectionError):
+            pass
+    return decorated
+
+
+
```
Comment (dimagimon) [inline testapps/test_pillowtop/tests/utils.py]: W391 blank line at end of file

File: testapps/test_pillowtop/tests/utils.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import functools
+from elasticsearch import exceptions as elasticsearch_exceptions
+from requests import exceptions as requests_exceptions
+
+
+def require_elasticsearch(fn):
```
Comment (czue) [inline testapps/test_pillowtop/tests/utils.py]: i suspect i may end up deleting this soon - planning on setting up ES-specific tests

**Comment by czue** [comment]
@dannyroberts I think this is g2g (pretty small changes - and I'd like to get them in before adding more)

goes with https://github.com/dimagi/fluff/pull/102 and https://github.com/dimagi/pillowtop/pull/147

**Pull request merged**